dojo.py

```diff
@@ -305,7 +306,116 @@ def load_people(self, filename):
             return "Sorry that file does not exist"
 
     def save_state(self, db_name=None):
-        pass
+        """ This method saves data from app into database"""
+
+        if db_name:
+            db = db_name + '.db'
+            engine = create_engine('sqlite:///{}'.format(db))
+        else:
+            engine = create_engine('sqlite:///dojo.db')
+
+        Base.metadata.create_all(engine)
+        session = sessionmaker(bind=engine)
+        new_session = session()
+
+        # loop through all rooms
+        for space in self.total_rooms:
+            room = Room()
+
+            room.room_name = space.room_name
+            room.room_type = space.type
+            room.maximum_occupants = space.maximum_occupants
+            room.current_occupants = len(space.occupants)
+
+            new_session.add(room)
+            new_session.commit()
+
+        # loop through all people
+        for individual in self.total_people:
+            person = People()
+
+            person.name = individual.person_name
+            person.role = individual.role
+            person.accomodation = individual.accomodation
+
+            for space in self.total_rooms:
+                if individual in space.occupants and space.type == "office":
+                    person.office_allocated = space.room_name
+                if individual in space.occupants and space.type == "living":
+                    person.living_space_allocated = space.room_name
+
+            new_session.add(person)
+            new_session.commit()
+
+        return 'You have saved data to the database'
```

Comment:
There is chance to duplicate database entries for one that has previously got some information, I would suggest you have a filter statement somewhere in your lines before adding new person or room where you make sure the new item added isn't similar to one that already exists